servlet/src/main/java/org/togglz/servlet/activation/ClientIpActivationStrategy.java

```diff
@@ -39,28 +45,49 @@ public String getName()
    @Override
    public boolean isActive(FeatureState featureState, FeatureUser user)
    {
-
       HttpServletRequest request = HttpServletRequestHolder.get();
       if (request != null) {
 
-         String allowedIpsParam = featureState.getParameter(PARAM_IPS);
-         List<String> allowsIps = Strings.splitAndTrim(allowedIpsParam, "[\\s,]+");
-
-         // TODO: This should support a simple form of subnet matching
-         return allowsIps.contains(request.getRemoteAddr());
-
+         List<String> parts = Strings.splitAndTrim(featureState.getParameter(PARAM_IPS), "[\\s,]+");
+
+         List<InetAddress> ips = new ArrayList<>();
```

Comment:
I don't think that you have to build these lists first and validate in a separate code block. Why not doing this in one step. This would even be more efficient, because you have a result as soon as the first entry matches and therefore don't have parse the following addresses.